sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -1048,6 +1048,9 @@ object TypeCoercion extends TypeCoercionBase {
           }
         }
 
+      // Allows the cast between different collated strings to match with ANSI behavior.
```

Comment:
This is a behavior change.

@uros-db I looked at the code base, we have implicit cast rules for `AbstractArrayType`, but not `AbstractMapType`, seems like a miss?

Reply:
`AbstractMapType(StringTypeAnyCollation, StringTypeAnyCollation)` does not work with non ANSI type coercion here. We should fix it here in any event.

Reply:
@uros-db please open a separate PR - I can close mine.

Reply:
on it